permissions/resource_permissions.go

```diff
@@ -176,6 +174,21 @@ func (a PermissionsAPI) Update(objectID string, objectACL AccessControlChangeLis
 			PermissionLevel: "CAN_MANAGE",
 		})
 	}
+	w, err := a.client.WorkspaceClient()
+	if err != nil {
+		return err
+	}
+	if strings.HasPrefix(objectID, "/sql/warehouses") {
+		warehouse, err := w.Warehouses.GetById(a.context, strings.ReplaceAll(objectID, "/sql/warehouses/", ""))
+		if err != nil {
+			return err
+		}
+		// add CAN_MANAGE permission for creator
+		objectACL.AccessControlList = append(objectACL.AccessControlList, AccessControlChange{
+			UserName:        warehouse.CreatorName,
```

Comment:
will it work if it's a service principal? check `azure-prod` env

Reply:
our API accepts service principal as UserName fine...which is the cause of #2131